### ISSUE TITLE
Can't access tailnet domain

### ISSUE THREAD

**Issue by excelsior091224** [tailscale/tailscale#5541]
### What is the issue?

After updating to version 1.30.0, I cannot access machines in the tailnet using the domain set up with [HTTPS Certificates](https://tailscale.com/kb/1153/enabling-https/).

### Steps to reproduce

updateing to 1.30.0

### Are there any recent changes that introduced the issue?

updateing to 1.30.0

### OS

Windows

### OS version

Windows 11 Home 21H2

### Tailscale version

1.30.0 (t7fd4d617a-g7395c0dee)

### Bug report

BUG-4aa8af858d56268789e42bc4113ec8d7df857b74f2c64de1789f4e19ae8e3f7c-20220903222730Z-851141de0f684425

**Comment by DentonGentry** [comment]
What happens when you try?
- it says the name is not resolved?
- it resolves the name but cannot connect?
- it can connect but gets an error?
- something else?

**Comment by excelsior091224** [comment]
`ERR_NAME_NOT_RESOLVED`

If I try to exit tailscale and restart, it is able to access.
But, I restart Windows machine, it happens again.

**Comment by DentonGentry** [comment]
Indeed, around the time of the bugreport we don't see the TCD being sent:
`2022-09-04 07:41:58.5897086 +0900 +0900: dns: Resolvercfg: {Routes:{.:[...]} Hosts:9 LocalDomains:[beta.tailscale.net. example.com.beta.tailscale.net. ts.net.]+65arpa}`

There is no `tails-scales.ts.net` in the list.

A number of TCD-related changes went in to making MagicDNS Generally Available: https://tailscale.com/blog/magicdns/
What does it do now?

**Comment by excelsior091224** [comment]
I still frequently have trouble accessing it.
I checked and the DNS address displayed by the nslookup command is not 100.100.100.100 and MagicDNS does not seem to be working.
Each time this happens, I disable MagicDNS in the admin console and then re-enable it.

**Comment by excelsior091224** [comment]
There are no other VPN and only Windows Defender in my pc.
